documentation/modules/cruise-control/con-rebalance-performance.adoc

```diff
@@ -0,0 +1,53 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='con-rebalance-{context}']
+
+= Rebalance performance tuning overview 
+
+When an xref:con-optimization-proposals-{context}[optimization proposal] is xref:proc-approving-optimization-proposal-{context}[approved], the partition reassignment commands that comprise the proposal are applied to the Kafka cluster by the Cruise Control server. 
+These commands consist of two types of operations:
+
+* Partition movement: Involves transferring the partition replica and its data to a new location. Partition movements can take one of two forms:
+    ** Inter-broker movement: The partition replica is moved to a log directory on a different broker.
+    ** Intra-broker movement: The partition replica is moved to a different log directory on the same broker.
+
+* Leadership movement: This involves switching the leader of the partition's replicas.
+
+Cruise Control batches up these reassignment commands and issues them to the Kafka cluster. 
+The performance of the cluster during the rebalance is affected by the quantities of each type of movement contained in each batch.
+Cruise Control provides several configuration options for tuning a cluster rebalance for optimal performance. 
+
+You can set these _tuning_ options at either the xref:ref-cruise-control-configuration-{context}[Cruise Control server] or xref:proc-generating-optimization-proposals-{context}[optimization proposal] levels:
+
+* The Cruise Control server setting can be set in the Kafka custom resource under `Kafka.spec.kafka.spec.cruiseControl.spec`. 
+* The individual rebalance performance configurations can be set under `KafkaRebalance.spec`. 
+
+The relevant configurations are summarised below:
+
+|==========================================================================================================================
+| Server Configuration                              | KafkaRebalance Configuration            | Description | Default Value 
+
+| `num.concurrent.partition.movements.per.broker`   | `concurrentPartitionMovementsPerBroker` | 
+  The maximum number of inter-broker partition movements in each partition reassignment batch | 10
+
+| `num.concurrent.intra.broker.partition.movements` | `concurrentIntraBrokerPartitionMovements`     | 
+  The maximum number of intra-broker partition movements in each partition reassignment batch | 2
```

Comment:
I'm surprised by this default. Perhaps naively I would have expected that since intrabroker movement is "just copying some files around" (i.e. not requiring any network) it would be less burdensome on the cluster and therefore the default could be higher than the interbroker case.Is my intuition wrong, or is it just a weird default?

Reply:
So having dug into the code and the inter-broker default is not 10 (as it says in the CC config docs - I have submitted an upstream PR to fix that) it is actually 5 which makes the difference compared to the intra-broker value less. But, I agree this seems counter-intuitive and it is not clear why it is lower than the inter-broker value.